### ISSUE TITLE
`sys.version_info` check as the the right operand

### ISSUE THREAD

**Issue by Viicos** [microsoft/pyright#11241]
On 1.1.408:

Code sample in [pyright playground](https://pyright-play.net/?pythonVersion=3.9&code=JYWwDg9gTgLgBAZwJ4IFCkrOMlgKZqqrTADmwAdgPo75wC8cAdC0cAGaIpMBueUCYBGqV2EOAD5GACgDMAGjgBGAAwBKOAEMKAEzgly1WnjjAE2XASYBVCkIoAVSwC44cAMT6A1qjduwmgiEHHByiqoaADyMyAi8-ILCVKLi2noGlDSWpubGcbb2TviuHnD8UNC%2BfgFBRKieAIIAxjAArpoANnCtCCZNgSZmzuicYcrqcNFccXwC9skUYpOhCsoALBpp%2BlBkmcY5Fvj5dsJFeCWe5ZV%2BcDVoQA)

```python
import sys
import types


origin_type = ...

if sys.version_info >= (3, 10) and origin_type is types.UnionType:  # ok
    pass

if (3, 10) <= sys.version_info and origin_type is types.UnionType:  # error
    pass


# Actual use case is:
if (3, 10) <= sys.version_info < (3, 14) and origin_type is types.UnionType:  # error
    pass

```

**Comment by erictraut** [comment]
This is by design, not a bug. The Python typing spec is clear about which forms a type checker must honor for system version checks. These need to be performed early in the analysis process before type evaluation and code flow are possible (because they inform the code flow graph). That means the type checker needs to use basic AST "shape" comparisons when evaluating these expressions. If you use an unsupported "shape", it won't work.
